src/test/java/com/znsio/sample/e2e/businessLayer/amazonsearch/AmazonSearchBL.java

```diff
@@ -0,0 +1,50 @@
+package com.znsio.sample.e2e.businessLayer.amazonsearch;
+
+import com.context.TestExecutionContext;
+import com.znsio.e2e.entities.Platform;
+import com.znsio.e2e.runner.Runner;
+import com.znsio.sample.e2e.entities.SAMPLE_TEST_CONTEXT;
+import com.znsio.sample.e2e.screen.amazonsearch.AmazonHomeScreen;
+import org.apache.log4j.Logger;
+import org.assertj.core.api.SoftAssertions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AmazonSearchBL {
+
+    private static final Logger LOGGER = Logger.getLogger(AmazonSearchBL.class.getName());
+    private final TestExecutionContext context;
+    private final SoftAssertions softly;
+    private final String currentUserPersona;
+    private final Platform currentPlatform;
+
+    public AmazonSearchBL(String userPersona, Platform onPlatform) {
+        long threadId = Thread.currentThread().getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        this.currentUserPersona = userPersona;
+        this.currentPlatform = onPlatform;
+        Runner.setCurrentDriverForUser(userPersona, onPlatform, context);
+    }
+
+    public AmazonSearchBL() {
+        long threadId = Thread.currentThread().getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        this.currentPlatform = Runner.platform;
+    }
+
+    public AmazonSearchBL searchForProduct(String productTitle) {
+        int numberOfProductsFound = AmazonHomeScreen.get().searchForProductInSearchBar(productTitle).getNumberOfProductsFound();
```

Comment:
variable name can be availableProducts

Reply:
NumberOfProductsFound depicts how many products were found with matching description on the screen as searching iPhone 13 might also list iPhone 12 of 14 too.